### ISSUE TITLE
Rserve error: unknown variable / method

### ISSUE THREAD

**Issue by dlebauer** [ebimodeling/ghgvc#117]
I get this on ebimodeling/ghgvc master branch when I do the following

1. Follow docker deploy commands in README
2. Click a bunch of places
3. Run the app

![8ba59bb0-3b9d-4c9e-9d0d-20ad31f4c11f](https://user-images.githubusercontent.com/464871/42861434-cb716016-8a0f-11e8-96ab-5f9c2231f39f.png)

The error text includes:

Completed 500 Internal Server Error ...

Rserve::Connnection::EvalError ... status=error unknown variable/method

**Comment by teixeirak** [comment]
@caseyhofford, is this where its currently stalling?

**Comment by caseyhofford** [comment]
@teixeirak yes, this is the issue that causes some locations to never load.

**Comment by caseyhofford** [comment]
![bug 117 detailed output](https://user-images.githubusercontent.com/8454154/47942225-cd06f700-deb6-11e8-9b1a-8906fa39b569.PNG)

The traceback here shows that the error is happening in ghgvcr, however I can't find the offending line of code.

`ghgvcr_1     | Error in if (string == "") x <- NA else if (string == "NA") x <- NA else if (string ==  :
ghgvcr_1     |   missing value where TRUE/FALSE needed
ghgvcr_1     | Calls: <Anonymous> -> lapply -> FUN
`

**Comment by dlebauer** [comment]
Is it here: https://github.com/ebimodeling/ghgvcR/blob/master/R/helpers.R#L256 ?

**Comment by dlebauer** [comment]
Are you running an interactive terminal, and if so, can you try 

```r
debugonce('str2LogicalOrNumeric')
```

This will put you inside the function when it is evaluated before the error occurs.
? 

Or, can you provide the output of `traceback()` after the error?

**Comment by teixeirak** [comment]
Is the problem as simple as needing to move the x inside the quotes?

**Comment by caseyhofford** [comment]
I was not using an interactive terminal but I will do that.
I need to clone ghgvcr on its own. I've just been debugging from the docker-compose image.
I was looking for any strange stuff in the JSON object being sent to R, but I will test those JSON objects directly in ghgvcr with the R debugger.

**Comment by caseyhofford** [comment]
Not sure about x in quotes but I will try playing with the syntax of that line while debugging.

**Comment by dlebauer** [comment]
OK here is a minimal working example of the error:

```r
z <- function(string){
   if (string == "") x <- NA
   else if (string == "NA") x <- NA
}
> z(NA)
Error in if (string == "") x <- NA else if (string == "NA") x <- NA else if (string ==  : 
  missing value where TRUE/FALSE needed
```

This works:

```r
zz <- function(string){
  if (is.na(string)) x <- NA
  else if (string == "") x <- NA
}
zz(NA)
```

**Comment by dlebauer** [comment]
See if https://github.com/ebimodeling/ghgvcR/pull/47 fixes this issue ...

**Comment by teixeirak** [comment]
Is this still a problem?

**Comment by caseyhofford** [comment]
@teixeirak yes, this issue is resolved with PR #47 . I tried approving the PR, but it doesn't seem to count my approval as meeting the requirements. Maybe if you approve it as well that PR can be completed? Or I may need to be added as a contributor and reviewer for the ghgvcR repository.

**Comment by teixeirak** [comment]
I merged and added permissions for you on the repository.
